File: cozumler/kelimeyi-tersten-yazma-problemi/javascript/kelimeyi-tersten-yazma-problemi.js
function reverse(str, index) {
  if ( str[index] ) {
    return reverse(str, index + 1) + str[index];
  } else {
    return "";
  }
}

const test_str = "TEST STRING KODLAMA PRATİKLERİ";
console.log( reverse(test_str , 0) );
// çıktı : İRELKİTARP AMALDOK GNIRTS TSET


/* ALTERNATİF YÖNTEM */

function reverse2(str){
  return str.split("").reverse().join("");
}

const test_str2 = "Bu bir reverse test";
console.log(reverse2(test_str2));
// çıktı: tset esrever rib ub

File: cozumler/kelimeyi-tersten-yazma-problemi/ocaml/kelimeyi-tersten-yazma-problemi.ml
let rec reverse str =
  if String.length str > 0 
  then 
  reverse (String.sub str 1 (String.length str - 1))
    ^
  String.sub str 0 1
  else str

let () =
  let test_string = "TEST STRING KODLAMA PRATIKLERI" in
  Printf.printf "%s" (reverse test_string)
  (* çıktı : IRELKITARP AMALDOK GNIRTS TSET *)
